### DISCUSSION TITLE
Fritz!Box(en) Maximale Blocklist-Größen

### DISCUSSION THREAD

**Discussion by Enkidu70** [haumacher/phoneblock#63] [Q&A]
Was ist die maximale Blocklist-Größe für welches Fritz!Box Modell?

Wenn Dein Modell bereits gelistet ist, bitte antworte auf den ersten Beitrag zum Modell.

**Comment by Enkidu70** [comment]
Fritz!Box 6591 Cable

**Reply by Enkidu70** [reply]
Meine FB 6591 zeigt mir bei 5000 67 Seiten zum Drucken an.
Bei 6000 ändert sich dieser Wert nicht mehr.

Abgestürzt ist sie zum Glück nicht. :)

**Comment by haumacher** [comment]
FRITZ!Box 7590

41 Seiten bei 3000 Nummern

**Reply by stephen-lockett** [reply]
Ich komme bei einem aktuellen Test mit dem gleichen Modell immer auf 20 Seiten.
Ich habe alles von 1.000 bis 6.000 Einträgen ausprobiert. D.h. die Blocklist-Größe scheint bei mir keine Rolle zu spielen.

Ich könnte mir vorstellen, dass die Liste mit der neuesten Firmware immer bei 1.000 Einträgen abgeschnitten wird.
FRITZ!OS-Version: 7.57

**Reply by haumacher** [reply]
Das kann ich nicht bestätigen. Habe auch FRITZ!Box 7590 mit FRITZ!OS:7.57 und komme aber auf 60 Seiten. Vielleicht passt was mit dem Update nicht? Nachdem man auf der PhoneBlock-Einstellungsseite die Größe aktualisiert hat, muss man in der Fritz!Box beim Telefonbuch den Update-Knopf drücken und das Update darf keine Fehlermeldung liefern. Wo auch immer die 41 Seiten aus der Orignal-Messung herkamen...

Hast Du vielleicht noch andere große Telefonbücher importiert, die den Speicherplatz beanspruchen?

**Reply by stephen-lockett** [reply]
Genau, ich bin dort auf Update gegangen, das lief ohne Fehler durch. Trotzdem danke für den Hinweis. ;)

Tatsächlich habe ich noch einige andere Adressbücher importiert, die relativ groß sind (wenn auch deutlich kleiner als 1.000 Einträge).
Vermutlich wird es dann tatsächlich an den weiteren Adressbüchern liegen.

**Update:**

Es sind insgesamt **6** zusätzliche Adressbücher mit insgesamt **278** Einträgen.
Das spricht dann eher dagegen, dass es an den zusätzlichen Adressbüchern liegt.

**Reply by stephen-lockett** [reply]
Ich habe gerade mal nachgeschaut, wie viele Kontakte (nicht Druckseiten; manche Kontakte haben mehrere Nummern) ich denn überhaupt in dem PhoneBlock-Adressbuch habe: **401**
Vielleicht ergeben sich dann ganz einfach nicht mehr Einträge mit den nachfolgenden Einstellungen?

![image](https://github.com/haumacher/phoneblock/assets/150799120/c21d6f37-9a34-4511-915c-d8fbd1be3cca)

Ansonsten habe ich gerade keine Idee mehr, woran es sonst noch liegen könnte...

**Reply by haumacher** [reply]
Die Nummern werden so zusammengefasst, dass pro Kontakt maximal 9 Nummer verwendet werden (das Fritz!Box-Limit). Bei 401 Kontakten könnten das ca 3000 Nummern sein, ja. Vielleicht liegt es an den Drucker-Einstellungen, dass die Anzahl der Seiten nicht vergleichbar ist?

**Reply by stephen-lockett** [reply]
Ah, das klingt plausibel. Danke für die Erklärung!
Das mit den Druckereinstellungen könnte sich tatsächlich auch so verhalten.

**Comment by Thirteen-37** [comment]
Ich habe da auch mal eine vielleicht blöde Frage, aber:
Auf der phoneblock Seite, kann ich in den Einstellungen ja die Maximale Blocklist-Größe auswählen (bis zu 6.000 Einträge). Auf der Status Seite steht, dass "12.320 aktive Nummern auf der Blocklist" sind.
Könnten vielleicht die 'Erfinder' der Liste, diese nicht einfach stückeln?
Wenn z.B. bei einer Fritzbox 7590 ca. 3.000 Einträge funktionieren, könnte man doch vielleicht einfach 5 Listen anlegen und dann sollten alle im Speicher sein.
Natürlich müsste man dafür auch 5 verschiedene CardDAVs machen bzw. noch mehr, wenn die Liste weiter wächst.

Oder machen die Fritzboxen generell bei vielen Einträgen Probleme, ganz unabhängig davon, auf wie viele Listen diese verteilt sind?

**Reply by haumacher** [reply]
Das Stückeln der Liste ist technisch nicht ganz einfach und hat nur vage Erfolgsaussichten. Schon mit 3000 Einträgen wird die Fritz!Box-Oberfläche merklich träger. Meine Befürchtung ist, dass ein Aufteilen der Liste trotzdem die Box an ihre Grenzen bringt. Daher glaube ich nicht, dass sich der Aufwand lohnen würde das auszuprobieren. Zumal die CardDAV-Implementierung in der Fritz!Box gelinde gesagt volldeppert ist - die Box macht immer einen Komplettabruf der Liste, obwohl man mit CardDAV eigentlich nur die Änderungen abfragen kann, die Box akzeptiert keine Kompression und lässt sich für jede einzelne Nachricht erneut um ein Passwort bitten. Daher würde sich bei Verdreifachung der Liste auch die Serverlast und die erforderliche Netzwerkbandbreite verdreifachen.
